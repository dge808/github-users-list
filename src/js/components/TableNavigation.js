import React from 'react';

const TableNavigation = ({ onPageChange, currentPage, perPage, onPerPageChange }) => {
    const isClickForbidden = (currentPage - 1) < 0 ? {cursor: 'not-allowed'} : null;
    return (
        <div class="table-navigation">
          <span class="navigation-button"
                style={isClickForbidden}
                onClick={() => onPageChange(currentPage - 1)}>
              &lt; &lt; Previous
          </span>
            <p class="current-page-number">
                Current Page: {currentPage + 1}
            </p>
            <span class="navigation-button"
                  onClick={() => onPageChange(currentPage + 1)}>
              Next &gt; &gt;
          </span>

            <p>Users per page:
                <select value={perPage} onChange={onPerPageChange}>
                    <option value={10}>10</option>
                    <option value={25}>25</option>
                    <option value={50}>50</option>
                    <option value={100}>100</option>
                </select>
            </p>
        </div>
    )
};

export default TableNavigation;

TableNavigation.propTypes = {
    onPageChange: React.PropTypes.func,
    currentPage: React.PropTypes.number,
    perPage: React.PropTypes.number,
    onPerPageChange: React.PropTypes.func
};