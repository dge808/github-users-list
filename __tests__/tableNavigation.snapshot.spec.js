import React from 'react';
import renderer from 'react-test-renderer';
import TableNavigation from '../src/js/components/TableNavigation';

describe('Table Navigation(snapshot)', () => {
    it('should match Table Navigation snapshot', () => {
        const component = renderer.create(<TableNavigation currentPage={1} perPage={25}/>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    })
});