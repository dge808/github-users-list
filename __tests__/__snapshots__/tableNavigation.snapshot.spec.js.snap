exports[`Table Navigation(snapshot) should match Table Navigation snapshot 1`] = `
<div
  className="table-navigation">
  <span
    className="navigation-button"
    onClick={[Function]}
    style={null}>
    < < Previous
  </span>
  <p
    className="current-page-number">
    Current Page: 
    2
  </p>
  <span
    className="navigation-button"
    onClick={[Function]}>
    Next > >
  </span>
  <p>
    Users per page:
    <select
      onChange={undefined}
      value={25}>
      <option
        value={10}>
        10
      </option>
      <option
        value={25}>
        25
      </option>
      <option
        value={50}>
        50
      </option>
      <option
        value={100}>
        100
      </option>
    </select>
  </p>
</div>
`;
